gui.py

```diff
@@ -233,12 +281,25 @@ def initUI(self, song):
         for init_cmd in self.device.init_command:
             self.queue_out.put(init_cmd)
 
+        self.updatePorts()
         self.update()
 
+    def updatePorts(self):
```

Comment:
Le 15/06/2015 12:53, Julian Jarecki a écrit :
>
> In gui.py 
> <https://github.com/Vampouille/superboucle/pull/22#discussion_r32408825>:
>
> >          self.update()
> >
> > +    def updatePorts(self):
>
> Sorry, not sure what you mean, could you explain?
>

This is where you can put most of example code :

* retreive new port name from song : [clip.output for clip in song.clips]
* unregister no more needed ports (bass and piano in example)
* create new port (delay in example)

after updatePorts() call jack client should have outputs corresponding 
to each clip.output value

With this code, output ports with same name won't be deleted and 
recreated when you load new song. so you won't have to reconnect ports 
with jack routing, it can be useful for live performance.

You should also add special if for 'MAIN' output, this output should not 
be deleted.